core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -117,11 +117,28 @@ class KafkaApis(val requestChannel: RequestChannel,
         case ApiKeys.DELETE_RECORDS => handleDeleteRecordsRequest(request)
         case ApiKeys.INIT_PRODUCER_ID => handleInitProducerIdRequest(request)
         case ApiKeys.OFFSET_FOR_LEADER_EPOCH => handleOffsetForLeaderEpochRequest(request)
-        case ApiKeys.ADD_PARTITIONS_TO_TXN => handleAddPartitionToTxnRequest(request)
-        case ApiKeys.ADD_OFFSETS_TO_TXN => handleAddOffsetsToTxnRequest(request)
-        case ApiKeys.END_TXN => handleEndTxnRequest(request)
-        case ApiKeys.WRITE_TXN_MARKERS => handleWriteTxnMarkersRequest(request)
-        case ApiKeys.TXN_OFFSET_COMMIT => handleTxnOffsetCommitRequest(request)
+        case ApiKeys.ADD_PARTITIONS_TO_TXN => handleWithInterBrokerVersionAtLeast(KAFKA_0_11_0_IV0, request, (error:Errors) => {
```

Comment:
`ApiKeys` could have a `minMagic` or `minMessageFormatVersion` field. By default there is no minimum (which is the common case). For the ones that have a minimum, they can use an overloaded constructor to indicate that. Maybe we should change the constructor that takes `clusterAction` to have an additional parameter and add a constructor that takes id and name.